ili/inference/runner_lampe.py

```diff
@@ -0,0 +1,333 @@
+            likelihood models, or ratio classifiers
+        engine (str): name of the engine class (NPE only)
+        train_args (Dict): dictionary of hyperparameters for training
+        out_dir (Path): directory where to store outputs
+        device (str): device to run on
+        proposal (Distribution): proposal distribution from which existing
+            simulations were run, for single round inference only. By default,
+            we will set proposal = prior unless a proposal is specified.
+        name (str): name of the model (for saving purposes)
+        signatures (List[str]): list of signatures for each neural net
+    """
+
+    def __init__(
+        self,
+        prior: Distribution,
+        nets: List[Callable],
+        engine: str = 'NPE',
+        train_args: Dict = {},
+        out_dir: Path = None,
+        device: str = 'cpu',
+        proposal: Distribution = None,
+        name: Optional[str] = "",
+        signatures: Optional[List[str]] = None,
+    ):
+        self.prior = prior
+        self.nets = nets
+        if engine != 'NPE':
+            logging.warning(
+                'lampe only supports NPE engine. Engine set to NPE.')
+        self.engine = 'NPE'
+        self.train_args = dict(
+            training_batch_size=50, learning_rate=5e-4,
+            stop_after_epochs=30, clip_max_norm=5,
+            max_epochs=int(1e10),
+            validation_fraction=0.1)
+        self.train_args.update(train_args)
+        self.out_dir = out_dir
+        if self.out_dir is not None:
+            self.out_dir = Path(self.out_dir)
+            self.out_dir.mkdir(parents=True, exist_ok=True)
+        self.device = device
+        if proposal is None:
+            self.proposal = prior
+        else:
+            self.proposal = proposal
+        self.name = name
+        self.signatures = signatures
+        if self.signatures is None:
+            self.signatures = [""]*len(self.nets)
+
+    @classmethod
+    def from_config(cls, config_path: Path, **kwargs) -> "LampeRunner":
+        """Create a lampe runner from a yaml config file
+
+        Args:
+            config_path (Path, optional): path to config file
+            **kwargs: optional keyword arguments to overload config file
+        Returns:
+            LampeRunner: the lampe runner specified by the config file
+        """
+        with open(config_path, "r") as fd:
+            config = yaml.safe_load(fd)
+
+        # optionally overload config with kwargs
+        config.update(kwargs)
+
+        # load prior distribution
+        config['prior']['args']['device'] = config['device']
+        prior = load_from_config(config["prior"])
+
+        # load proposal distributions
+        proposal = None
+        if "proposal" in config:
+            config['proposal']['args']['device'] = config['device']
+            proposal = load_from_config(config["proposal"])
+
+        # load embedding net
+        if "embedding_net" in config:
```

Comment:
Please test this works in the unit tests

Reply:
Added.